### ISSUE TITLE
Segfault of Tarantool 1.6.7 on FreeBSD

### ISSUE THREAD

**Issue by stofel** [tarantool/tarantool#1231]
OS FreeBSD10.2 amd64
tarantul Tarantool 1.6.6-130-ge3f2420 (Lua console)
init lua:

```lua
box.cfg{listen = '127.0.0.1:3301'}
local console = require('console')
console.listen('127.0.0.1:3312')
fiber = require('fiber')
s = box.space.test
if not s then
  s = box.schema.create_space('test', { id = 0 })
  s:create_index('primary', {type = 'tree', id = 0})
  for i = 1, 10 do
    fiber.sleep(1)
    print('insert ' .. i)
    s:insert{i, 'my_app tuple'}
  end
end
```
lua hello world:
```lua
function hello()
  return 'hello'
end
```
Try it in console:

``` console
$ rlwrap nc localhost 3312
tarantool> dofile('hello.lua')
---
...
tarantool> hello()
---
- hello
...
```

Try python
```python
#!/usr/local/bin/python
import tarantool

server = tarantool.connect("localhost", 3301)
demo = server.space(0)
R = demo.call("hello", [])
print R
```

./call_hello.py
tarantool.error.NetworkError: (54, 'Connection reset by peer')
segfault  tarantul and in tarantool log:

```
2015-12-25 11:17:49.207 [46551] main/107/console/127.0.0.1:24604 I> client 127.0.0.1:24604 connected
Segmentation fault
Current time: 1451031588
Please file a bug at http://github.com/tarantool/tarantool/issues
Attempting backtrace... Note: since the server has already crashed, 
this may fail as well
```

**Comment by rtsisyk** [comment]
I can't reproduce this problem on the latest 1.6.7:
```
>>> R = demo.call("hello", [])
>>> print R
- [hello]

>>> server.eval("return box.info.version")
[1.6.7-557-gb8fdd95]

FreeBSD tt 10.2-RELEASE FreeBSD 10.2-RELEASE #0 r286666: Wed Aug 12 15:26:37 UTC 2015     root@releng1.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC  amd64

```

1.6.7 is buggy when compiled by clang. We substantially improved support of FreeBSD forupcoming 1.6.8 release. Please compile 1.6.7 (branch `master`) using gcc or switch to 1.6.8 (branch `1.6`.)

**Comment by stofel** [comment]
Yes, 1.6.7 from git (branch master) is work.
Seems it is bug only freebsd database/tarantool.

**Comment by rtsisyk** [comment]
@zloidemon could you please update FreeBSD/ release?

**Comment by rtsisyk** [comment]
Most problems with FreeBSD have been fixed in 1.6.8. @zloidemon  needs a source tarball to update FreeBSD ports. @stofel please wait https://github.com/tarantool/build/issues/5 to be closed and then ping @zloidemon or just compile tarantool from git (branch 1.6).
